### PR TITLE
Added Custom Annotations support in crdb cluster #696

### LINKED ISSUE #696: Implement Custom Annotations in CR
Provide functionality in the CR to the end user to be able to annotate clusters  for 3rd party service integration

### PR DESCRIPTION
**Pull request by abhishekdwivedi3060**
Closes #696 

* Added Custom Annotations support in crdb cluster
* Added the new API field "additionalAnnotations" for adding user defined annotations to CRDB cluster resources
* Added/Modified existing unit test for this change

### CODE REVIEW

File: pkg/resource/statefulset.go [hunk 1]
```diff
@@ -191,6 +194,8 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 	pod := corev1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
 			Labels: b.Selector,
```
Comment (chrislovecnm) [inline pkg/resource/statefulset.go:196]: Does this cause pod restart? Does datadog, for instance, need annotations on the pods?

Comment (abhishekdwivedi3060) [reply pkg/resource/statefulset.go]: Yes, it does. Change it sts spec triggers restart.
Also, most of the monitoring tools like Datadog, Prometheus works on pod annotations only, so added custom annotations in the pod meta as well

File: apis/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -109,6 +109,10 @@ type CrdbClusterSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Map of additional custom labels"
 	// +optional
 	AdditionalLabels map[string]string `json:"additionalLabels,omitempty"`
+	// (Optional) Additional custom resource annotations that are added to all resources
```
Comment (pseudomuto) [inline apis/v1alpha1/cluster_types.go:112]: Think we should add something here to remind users that changing this will restart the cluster?

**Review by pseudomuto — DISMISSED** [review]

File: apis/v1alpha1/cluster_types.go (hunk 2 above)
Comment (abhishekdwivedi3060) [reply apis/v1alpha1/cluster_types.go]: Done

**Review by pseudomuto — APPROVED** [review]

**Pull request merged**